src/hooks/useCurrencies.ts

```diff
@@ -113,18 +113,18 @@ export const useCurrencies = () => {
 
       const { amount: conversionRate, decimals } = convertedCurrency;
       const convertAmount = (amount: string) => {
-        return conversionRate ? (conversionRate * stringToNumber(amount)).toString() : '';
+        return conversionRate
+          ? (conversionRate * stringToNumber(amount)).toFixed(decimals)
```

Comment:
The decimals should not be included here as we have a display utility function that handles decimals

Reply:
As discussed I will delete this PR and create a new one with much less impact on the code, but for the record, here my concern was that we should not store a price into a string without using the decimals, because it creates confusion between truncated amount strings and untrucated amount strings (I would have been ok with this flow  if the price was kept as a number until the conversion). So I decided to de the conversion here, and then to modify the conversion function to not reconvert the prices if they are already in strings.